test/garbage-test.js

```diff
@@ -0,0 +1,92 @@
+import { describe, it } from 'mocha'
+import assert from 'assert'
+
+import { resolve, reject, fulfill, never } from '../src/main'
+import getCounts, { collect, formatResults, sumResults } from './lib/refcount-util'
+
+describe('reference counts', () => {
```

Comment:
Can you give me the rundown in a bit more detail what this is testing, and what the Handle approach "fixes" or improves?  I see the words "reference counts", but this test and `refcount-util` are quite dense :)  I see it counting something and then asserting the count against a threshold, but it's difficult to understand exactly what's going on.

Thanks!